apps/designer/app/shared/fetcher/persistent-fetcher.tsx

```diff
@@ -0,0 +1,82 @@
+import { type FetcherWithComponents, useFetcher } from "@remix-run/react";
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useRef,
+  type ReactNode,
+} from "react";
+import { useOnFetchEnd } from "./on-fetch-end";
+
+/* This utility is designed for the case when you want to send a request using Remix's fetcher,
+ * but you don't want the request to be cancelled when the component where useFetcher() is called unmounts.
+ */
+
+type Task<Data> = {
+  target: Parameters<FetcherWithComponents<Data>["submit"]>[0];
+  options: Parameters<FetcherWithComponents<Data>["submit"]>[1];
+  callback?: (data: Data) => void;
+};
+
+type AddTaskFn = <Data>(
+  target: Task<Data>["target"],
+  options: Task<Data>["options"],
+  callback?: Task<Data>["callback"]
+) => void;
+
+const Context = createContext<AddTaskFn | undefined>(undefined);
+
+export const PersistentFetcherProvider = ({
```

Comment:
why do you need a provider for this vs a simple globally handled queue like we have in canvas/shared/sync/queue ?

Reply:
We need a global fetcher created using `useFetcher` which can be done only in a React component.

Reply:
right, fetcher can't be created outside